### PR TITLE
Cap maximum gas price

### PR DESCRIPTION
**Pull request by Longarithm**
Cap maximum gas price: https://github.com/near/nearcore/issues/4302

**Test plan**

- `test_cap_max_gas_price` checks that `max_gas_price <= 10 * min_gas_price` for the new protocol version.
- `process_blocks.rs:cap_max_gas_price_tests` check that `gas_price` may exceed `10 * min_gas_price` for the old protocol version but may not for the new one.

### CODE REVIEW

**Review by bowenwang1996 — CHANGES_REQUESTED** [review]
Please add a test that asserts the gas price does not exceed maximum gas price even when there is congestion.

**Comment by Longarithm** [comment]
Added new tests and moved creation of congestion to a separate function

File: chain/client/tests/process_blocks.rs [hunk 1]
```diff
@@ -77,6 +77,79 @@ pub fn create_nightshade_runtimes(genesis: &Genesis, n: usize) -> Vec<Arc<dyn Ru
         .collect()
 }
 
+/// Create environment and set of transactions which cause congestion on the chain.
+fn create_env_with_congestion(
```
Comment (bowenwang1996) [inline chain/client/tests/process_blocks.rs:83]: Nice

File: chain/client/tests/process_blocks.rs [hunk 2]
```diff
@@ -2745,6 +2761,45 @@ fn test_congestion_receipt_execution() {
...
+            env.produce_block(0, i);
+            let block = env.clients[0].chain.get_block_by_height(i).unwrap().clone();
+            let protocol_version = env.clients[0]
+                .runtime_adapter
+                .get_epoch_protocol_version(block.header().epoch_id())
+                .unwrap();
+            let min_gas_price =
+                env.clients[0].chain.block_economics_config.min_gas_price(protocol_version);
+            if block.header().gas_price() > 10 * min_gas_price {
+                return true;
```
Comment (bowenwang1996) [inline chain/client/tests/process_blocks.rs]: Please also assert that congestion actually happens. You can do this by asserting that gas usage is at least as large as gas limit.

**Review by bowenwang1996 — APPROVED** [review]
Well done!
